lesson_07/homeworks_07_panchenko.py

```diff
@@ -0,0 +1,137 @@
+# task 1
+""" Задача - надрукувати табличку множення на задане число, але
+лише до максимального значення для добутку - 25.
+Код майже готовий, треба знайти помилки та випраавити\доповнити.
+"""
+def multiplication_table(number):
+    multiplier = 1
+    while multiplier <= 10:  
+        result = number * multiplier
+        if result > 25:  
+            break  
+        print(f"{number} x {multiplier} = {result}")
+        multiplier += 1
+
+multiplication_table(3)
+
+# task 2
+"""  Написати функцію, яка обчислює суму двох чисел.
+"""
+
+def sum_of_two_numbers(a, b):
+    return a + b
+
+result = sum_of_two_numbers(4, 7)
+print(result)
+
+# task 3
+"""  Написати функцію, яка розрахує середнє арифметичне списку чисел.
+"""
+
+def average_of_list(numbers):
+    if len(numbers) == 0:
+        return 0 
+    return sum(numbers) / len(numbers)
+
+numbers = [5, 10, 15]
+average = average_of_list(numbers)
+print(average) 
+
+# task 4
+"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
+"""
+
+def reverse_string(input_string):
+    return input_string[::-1]
+
+reversed_str = reverse_string("hello")
+print(reversed_str)
+
+# task 5
+"""  Написати функцію, яка приймає список слів та повертає найдовше слово у списку.
+"""
+
+def longest_word(words):
+    if not words: 
+        return None
+    return max(words, key=len)
+
+words = ["apple", "banana", "grape"]
+longest = longest_word(words)
+print(longest)
+
+# task 6
+"""  Написати функцію, яка приймає два рядки та повертає індекс першого входження другого рядка
+у перший рядок, якщо другий рядок є підрядком першого рядка, та -1, якщо другий рядок
+не є підрядком першого рядка."""
+def find_substring(str1, str2):
+    return -1
+
+str1 = "Hello, world!"
+str2 = "world"
+print(find_substring(str1, str2))
+
+str1 = "The quick brown fox jumps over the lazy dog"
+str2 = "cat"
+print(find_substring(str1, str2))
+
+def find_substring(str1, str2):
+    return str1.find(str2)
```

Comment:
```suggestion
def find_substring(str1, str2):
    return str1.find(str2)
```
а тут все вірно, тільки нащо був код вище?